src/platforms/android/server/display.cpp

```diff
@@ -359,9 +359,9 @@ bool mga::Display::apply_if_configuration_preserves_display_buffers(
     if (!config.external().connected || displays.display_present(mga::DisplayName::external))
     {
         configure_locked(conf, lock);
-        return true;
+        return false;
```

Comment:
This is definitely wrong; it should return `true` if it applied the configuration. Otherwise the compositor will just call `configure` again.